src/main/java/org/folio/service/orders/flows/update/unopen/UnOpenCompositeOrderManager.java

```diff
@@ -189,17 +185,18 @@ private Future<Void> processInventoryOnlyWithItems(CompositePoLine compPOL, Requ
   }
 
   private Future<List<Piece>> deleteExpectedPieces(CompositePoLine compPOL, RequestContext requestContext) {
-    if (!PoLineCommonUtil.isReceiptNotRequired(compPOL.getReceiptStatus()) && Boolean.FALSE.equals(compPOL.getCheckinItems())) {
-      return pieceStorageService.getExpectedPiecesByLineId(compPOL.getId(), requestContext)
-        .compose(pieceCollection -> {
-          if (isNotEmpty(pieceCollection.getPieces())) {
-            return pieceStorageService.deletePiecesByIds(pieceCollection.getPieces().stream().map(Piece::getId).collect(toList()), requestContext)
-                                .map(v -> pieceCollection.getPieces());
-          }
-          return Future.succeededFuture(Collections.emptyList());
-        });
+    if (PoLineCommonUtil.isReceiptNotRequired(compPOL.getReceiptStatus()) && Boolean.TRUE.equals(compPOL.getCheckinItems())) {
```

Comment:
This change means that even if the receiving workflow is "Independent order and receipt quantity", pieces can be deleted when unopening an order. I don't see that change in logic reflected in the ticket or comments. Also the dialog mentions "Synchronized order and receipt quantity", so I imagine it could confuse users. Could you explain it ?

Reply:
But it is not: the opposite of
`!PoLineCommonUtil.isReceiptNotRequired(compPOL.getReceiptStatus()) && Boolean.FALSE.equals(compPOL.getCheckinItems())`
is:
`PoLineCommonUtil.isReceiptNotRequired(compPOL.getReceiptStatus()) || Boolean.TRUE.equals(compPOL.getCheckinItems())`

Reply:
thanks for this catch, initial logic was a bit hard to read, fixing

Reply:
Now it is the same as for creating/updating piece as should be:
https://github.com/folio-org/mod-orders/blob/master/src/main/java/org/folio/service/pieces/flows/strategies/ProcessInventoryStrategy.java#L56